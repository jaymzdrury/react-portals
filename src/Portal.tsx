import {createPortal} from 'react-dom'
import { useEffect } from 'react'

function Modal({children}:{children: React.ReactNode}) {
    
  const root = document.getElementById('portal')
  const element = document.createElement('div')

  useEffect(() => {
    root?.appendChild(element)
    return () => {root?.removeChild(element)}
  },[element, root])
  
  return createPortal(children, element)
}

export default Modal