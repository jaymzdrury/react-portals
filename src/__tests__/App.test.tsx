import { render } from '@testing-library/react';
import Portal from '../Portal'

test('portal renders when opened', () => {
  const {baseElement} = render(<Portal>Portal</Portal>)
  expect(baseElement).toMatchSnapshot()
})
