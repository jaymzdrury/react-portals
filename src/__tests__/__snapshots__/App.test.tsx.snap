// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal 1`] = `
<body>
  <div />
</body>
`;

exports[`portal renders when opened 1`] = `
<body>
  <div />
</body>
`;
